packages/graph-engine/src/nodes/string/split.ts

```diff
@@ -0,0 +1,41 @@
+import { NodeDefinition, NodeTypes } from "../../types.js";
+
+export const type = NodeTypes.SPLIT_STRING;
+
+export type State = {
+  string: string;
+  separator: string;
+};
+
+export const defaults: State = {
+  string: "",
+  separator: ",",
+};
+
+export const process = (input, state: State) => {
+    const final = {
+        ...state,
+        ...input,
+    };
+
+    if (final.separator === '') {
+        return [...final.string]
+    }
+
+    if (final.separator === undefined) {
+        return [final.string]
+    }
+
+    return final.string.split(final.separator)
+};
+
+export const mapOutput = (input, state, processed) => {
```

Comment:
This looks incorrect,at the end of process we have an array, we are then attempting to spread that array into an object for some reason? Lets just leave the default mapOutput so we have an output called `output` with the array for a user to manipulate?